main/lsp/lsp_helpers.cc

```diff
@@ -92,14 +92,15 @@ string LSPLoop::fileRef2Uri(const core::GlobalState &gs, core::FileRef file) {
 unique_ptr<Range> loc2Range(const core::GlobalState &gs, core::Loc loc) {
     unique_ptr<Position> start;
     unique_ptr<Position> end;
-    if (!loc.file().exists()) {
-        start = make_unique<Position>(1, 1);
-        end = make_unique<Position>(2, 0);
-    } else {
+    if (loc.exists()) {
         auto pair = loc.position(gs);
         // All LSP numbers are zero-based, ours are 1-based.
         start = make_unique<Position>(pair.first.line - 1, pair.first.column - 1);
         end = make_unique<Position>(pair.second.line - 1, pair.second.column - 1);
+    } else {
+        // this will happen if e.g. we disable the stdlib (e.g. to speed up testing in fuzzers).
```

Comment:
This seems like silent failure.
Can we make it _not_ return a range instead?

Reply:
@jvilk-stripe thoughts?

Reply:
update: After slack, we're going to make this return an optional, and in this branch we'll return nullopt.

Reply:
IRLed with @azdavis. We already guard calls to this with `loc.exists()`/`loc.file.exists()`, so I think it makes sense to make this and `loc2Location` return `optional<>`.